### ISSUE TITLE
Create Ads Module Setup CTA Banner

### ISSUE THREAD

**Issue by 10upsimon** [google/site-kit-wp#8660]
## Feature Description

With the Ads module now released, there is a need to create an Ads setup banner with applicable CTA that prompts the users to set up the module should they not yet have it set up. This setup banner will function like all prior setup banners and will contain a CTA to facilitate module set up, or be dismissed. 

The Figma design can be viewed [here](https://www.figma.com/file/THG1FJw5SaUxmiq38Mkf1x/Ads?type=design&node-id=850-1574&mode=design&t=pBBiRPBqxm3KgBLb-0). Following is a screenshot of the final setup banner design:

<img width="1680" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWUzQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--00c98c77499d596066d539aab9db3f9f948197f6/image.png" alt="image.png" />

Note, copy is not final at the time of writing, and will likely be altered prior to execution. This issue will be updated to reflect the final copy once it is approved.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* An Ads setup banner is to be created as per the Figma design [here](https://www.figma.com/file/THG1FJw5SaUxmiq38Mkf1x/Ads?type=design&node-id=850-1574&mode=design&t=pBBiRPBqxm3KgBLb-0).
* Said banner should only display for users who do not already have the Ads module set up
* Said banner should contain two CTA's as per the design:
    * A "Set up Ads" CTA button that takes the user to the Ads module setup screen
    * A "Maybe later" CTA link that dismisses the banner
* When the banner is dismissed via the "Maybe later" CTA link:
    * The banner should be hidden for 14 days
* When the banner re-appears after 14 days:
    * The "Maybe later" CTA should be renamed "Don’t show again"
    * Dismissing the banner via the "Don't show again" CTA should permanently hide/dismiss it for the user
 
## Implementation Brief

* [ ] Create a new component `assets/js/modules/ads/components/dashboard/AdsModuleSetupCTAWidget.js`
   * [ ] Add a new constant `ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY = "ads_module_setup_banner_prompt_dismissed_key"`
   * [ ] Use the `isModuleActive` selector on `CORE_MODULES` to check if the ads module is already active, if it is active, return null from the component so that it is not rendered on the dashboard.
   * [ ] Layout the component title, body and SVGs using CSS grid and flexbox in a SCSS file `assets/sass/components/ads/_googlesitekit-ads-setup-banner.scss` based on the latest [FIGMA design](https://www.figma.com/file/THG1FJw5SaUxmiq38Mkf1x/Ads?type=design&node-id=850-1574&mode=design&t=pBBiRPBqxm3KgBLb-0)
      * [ ] Implement the responsive single column layouts for [tablet](https://www.figma.com/file/THG1FJw5SaUxmiq38Mkf1x/Ads?type=design&node-id=1458-3296&mode=design&t=sbUar8fAu2ZfS1CC-0) and [mobile](https://www.figma.com/file/THG1FJw5SaUxmiq38Mkf1x/Ads?type=design&node-id=1458-2720&mode=design&t=5bNiSvelemdMcxJK-0).
      * [ ] Use the `useBreakpoint` hook to conditionally render the three different SVGs for each breakpoint; [desktop](https://www.figma.com/file/THG1FJw5SaUxmiq38Mkf1x/Ads?type=design&node-id=850-1574&mode=design&t=CKK4czQKT8lg2KSS-0), [tablet](https://www.figma.com/file/THG1FJw5SaUxmiq38Mkf1x/Ads?type=design&node-id=1458-3296&mode=design&t=sbUar8fAu2ZfS1CC-0) and [mobile](https://www.figma.com/file/THG1FJw5SaUxmiq38Mkf1x/Ads?type=design&node-id=1458-2720&mode=design&t=sbUar8fAu2ZfS1CC-0).
   * [ ] Create the primary CTA "Set up Ads" as a default `Button` component with which calls an `onSetup` callback function:
      * [ ] In this function, use the `activateModule` selector to get the `moduleReauthURL`, then use `navigateTo` to go to the module setup, [as we do in the `SetupModule` onSetup function](https://github.com/google/site-kit-wp/blob/e626cfc73ab79dd712b2fbc045fd8ae6db673cc2/assets/js/components/settings/SetupModule.js#L60-L69)
      * [ ] Use `trackEvent` here to track an `activate_module` event, [as we do in the `SetupModule` onSetup function](https://github.com/google/site-kit-wp/blob/e626cfc73ab79dd712b2fbc045fd8ae6db673cc2/assets/js/components/settings/SetupModule.js#L60-L69)
   * [ ] Use the [prompts functionality](https://github.com/google/site-kit-wp/blob/9555a2393b30a80b087ec9267b0bd629494f58cf/assets/js/googlesitekit/datastore/user/prompts.js) to conditionally render the entire component and the secondary CTA:
      * [ ] Get the dismissed state using `isPromptDismissed` passing the slug `ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY`.
      * [ ] Get the dismissed count using `getPromptDismissCount` passing the slug `ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY`.
      * [ ] If the prompt count is > 1 or the prompt is currently dismissed, return null from the component so that it doesn't render on the dashboard.
      * [ ] Add the secondary CTA using a tertiary button:
         * [ ] Conditionally render the Button text, if the dismissed count < 1, the text should be "Maybe later", otherwise it should be "Don't show again".
         * [ ] Create a `handleDismissClick` function and add it to the `onClick` event of this button
            * [ ] If the dismiss count is < 1, dismiss the prompt to the current timestamp + 14 days using: `await dismissPrompt( ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY, { expiresInSeconds: twoWeeksInSeconds});`
            * [ ] In the else statement, permanently dismiss the prompt by calling `dismissPrompt( ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY );` with no second argument.
 * [ ] Import and add the new component to the dashboard under the existing consent mode widget, rendering it conditional if the user is not in the view only context:
https://github.com/google/site-kit-wp/blob/9555a2393b30a80b087ec9267b0bd629494f58cf/assets/js/components/DashboardEntityApp.js#L225

### Test Coverage

* Create a new Storybook story for this component. Make sure each new story is added as a scenario to the VRTs by adding a scenario prop to each story.

## QA Brief

* Setup Site Kit
* Ensure Ads module is not active
* Verify that setup banner is showing (only when Ads module is not active)
* Verify that buttons behave as described in AC
* Once `Maybe Later` is clicked, banner should not show again. So, set the reference date in tester plugin to something over 14 days, 15days or 1 month for example. And verify that banner is showing again with second CTA having text "Don’t show again" and after it is clicked, banner is permanently dismissed

## Changelog entry

* Add the Ads module setup CTA banner.

**Comment by tofumatt** [comment]
Looks good. I'm guessing we don't have any GA events defined yet for this banner, but we should probably add those as well in a follow-up issue 😅 

(If we get them in soon we can probably just add it to this issue's ACs.)

ACs look good, moving to IB.

**Comment by benbowler** [comment]
I've created the IB today, one thing to note and discuss in the IBR is that we could quite easily have the Consent Mode and Ads Module widgets both rendering on the dashboard. We could prevent this as a one off by doing some kind of check for the Consent Mode display state within this new Widget but that's not very flexible and brings in CoMo logic to an Ads module widget, although perhaps this is acceptable for now?

We could perhaps create a follow up ticket to create a higher level component for these banners that handles the checks and only renders one at a time regardless if multiple widgets meet their display condition.

**Comment by benbowler** [comment]
Putting this into IBR, this comment from above still applies and is worth discussing:

> I've created the IB today, one thing to note and discuss in the IBR is that we could quite easily have the Consent Mode and Ads Module widgets both rendering on the dashboard. We could prevent this as a one off by doing some kind of check for the Consent Mode display state within this new Widget but that's not very flexible and brings in CoMo logic to an Ads module widget, although perhaps this is acceptable for now?
> 
> We could perhaps create a follow up ticket to create a higher level component for these banners that handles the checks and only renders one at a time regardless if multiple widgets meet their display condition.

**Comment by eugene-manuilov** [comment]
Thanks, @benbowler. Mostly looks good to me. 

> * Create the primary CTA "Set up Ads" as a default `Button` component which links to `${ settingsURL }&slug=ads&reAuth=true` with the `settingsURL` retrieved with the `getAdminURL` selector passing `googlesitekit-settings`.

We need to use the same approach that we use in the settings when we click on the connect button: https://github.com/google/site-kit-wp/blob/e626cfc73ab79dd712b2fbc045fd8ae6db673cc2/assets/js/components/settings/SetupModule.js#L60-L69

> ... we could quite easily have the Consent Mode and Ads Module widgets both rendering on the dashboard...

Yes, let's not worry about that for now. We will create a follow up task if it is needed.

**Comment by benbowler** [comment]
Thanks @eugene-manuilov, I've updated the IB module activation steps.

**Comment by eugene-manuilov** [comment]
Thanks, @benbowler. IB ✔️

**Comment by mohitwp** [comment]
### QA Update ❌ 

- Tested on dev environment.
- Verified all cases mentioned under AC and QAB.
- Compared design with Figma.

@zutigrm Please find my observations below

_1) Font family for the banner title is 'Google Sans Display' in Figma where as under actual implementation it is 'Google Sans Text'._

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/7afe015d-bd84-4eaf-88d4-96a3eadcbc7c)

![image](https://github.com/google/site-kit-wp/assets/94359491/9bfd226e-9e77-4adf-aefb-befb66df3b37)

</details>

_2)_ _We don't have figma design for large tablet viewports but design under Viewports having width between 961px to 1132px is looking off with space appearing below the image._ 

</details>

![image](https://github.com/google/site-kit-wp/assets/94359491/a848a1f6-e1f2-4f0a-a020-e2e86984a255)

![image](https://github.com/google/site-kit-wp/assets/94359491/c4727c2b-d576-4be1-8a4d-0ede2ed8e945)

![image](https://github.com/google/site-kit-wp/assets/94359491/e7743a74-3d67-491c-b25c-baf706815175)

![image](https://github.com/google/site-kit-wp/assets/94359491/69e0981e-844f-45b1-82e2-0471a41ca228)

</details>

_3) In smaller tablet viewports font size is different from figma. On smaller tablets we implemented same font size as mobile viewports. Tested on iPad air and Surface Pro smaller tablet viewports._

<details>

Figma -

![image](https://github.com/google/site-kit-wp/assets/94359491/5960d983-287d-4480-a9a6-ef4126e5195c)

![image](https://github.com/google/site-kit-wp/assets/94359491/24f0aea4-d198-4cb4-b970-cddc59e68fa5)

Dev environment -

![image](https://github.com/google/site-kit-wp/assets/94359491/c2cfdd49-04c0-412d-a4b7-fe350d8231f9)

![image](https://github.com/google/site-kit-wp/assets/94359491/17d40da9-19b8-41ee-9651-0055a285f1a2)

</details>

_4) The 'Maybe later' CTA is not functioning as expected according to the acceptance criteria and QA brief. Clicking on the 'Maybe later' CTA does not dismiss the banner but instead renames the CTA to "Don’t show again". Clicking on 'Don’t show again' dismisses the banner. Additionally, upon each refresh when the dashboard loads, the Ads module setup CTA banner briefly displays for a few milliseconds before hiding._

<details>


https://github.com/google/site-kit-wp/assets/94359491/2e3633c4-c921-455a-b2ea-c23802481d45

</details>

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

Tested on main environment.
Verified above reported issues  1 to 3 are resolved now.

@zutigrm I verified that on clicking the 'Maybe later' CTA banner, it gets dismissed and does not reappear when reloading the dashboard. However, it doesn't reappear even when I set the reference date in the tester plugin to over 14 days, 15 days, or 1 month later. Do I need to make some changes in the database, or is it getting permanently dismissed? Can you please check?

<details>

'Maybe later' CTA screencast

https://github.com/google/site-kit-wp/assets/94359491/f8689ab2-b5aa-4275-8088-83e99e878e1d


Banner not reappearing when I set the reference date in tester plugin to something over 14 days, 15days or 1 month.

https://github.com/google/site-kit-wp/assets/94359491/7b9e71f9-af0e-45d2-84c0-799a266b882f

</details>


**PASS CASES**


_1) Font family for the banner title is 'Google Sans Display' in Figma where as under actual implementation it is 'Google Sans Text'._ - **_PASS_**

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/3d3a6f9f-dfe1-454c-bc22-36cd40f25a3b)

</details>

2) We don't have figma design for large tablet viewports but design under Viewports having width between 961px to 1132px is looking off with space appearing below the image. - **_PASS_**

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/2c12b52a-23b8-4287-a79d-f62dcafe3d74)

![image](https://github.com/google/site-kit-wp/assets/94359491/9c6306ac-a47b-4fb0-bfb8-28717756fafe)

</details>

_3) In smaller tablet viewports font size is different from figma. On smaller tablets we implemented same font size as mobile viewports. Tested on iPad air and Surface Pro smaller tablet viewports._  **_- PASS_**

![image](https://github.com/google/site-kit-wp/assets/94359491/02fc678e-c161-4d28-a1e4-cebe90413ada)


![image](https://github.com/google/site-kit-wp/assets/94359491/89a6d2d8-70e0-444d-afb3-b2347476fbe6)

**Comment by zutigrm** [comment]
@mohitwp Thanks for checking it. Hm, I tested it manually on my end, as sometimes reference date doesn't affect some places. Which seems to be the case. You can edit the database, in `wp_usermeta` table, find the `wp_googlesitekitpersistent_dismissed_prompts` column and edit it.

<img width="582" alt="image" src="https://github.com/google/site-kit-wp/assets/10414808/2ea9d6b7-b8cb-404d-a7d0-1b8aee5b91be">

You will see `ads_module_setup_banner_prompt_dismissed_key` in the data, under it, there is `expires` property, you can replace current timestamp with this one `1716882640` (it is 28th May)

<img width="582" alt="image" src="https://github.com/google/site-kit-wp/assets/10414808/19170c58-d7e8-4624-9ebb-da856d91b395">

**Comment by mohitwp** [comment]
### QA Update ❌ 

- Tested on main environment.
- Verified functionality of 'Maybe later' CTA and 'Don't show' CTA.
- Note : Setting reference date in Tester plugin was not working so I tested it through db manipulation mentioned above.

@zutigrmI tested this on three different sites. According to the acceptance criteria, after clicking 'Maybe later', the CTA banner should display again in 14 days. In the database, the expiry date is correctly set to 14 days later. However, when I set a past date in the database, it forces the banner to display on the dashboard, but it still shows the 'Maybe later' CTA. It only displays the 'Don't show' CTA after appearing twice with the 'Maybe later' CTA.

Expected: On the second appearance, the banner should display the 'Don't show' CTA instead of 'Maybe later'.


<details>

https://github.com/google/site-kit-wp/assets/94359491/1dce9e42-911b-46f4-b66f-6dcaca5c13a3

</details>

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on main environment.
- Verified 'Maybe later' and 'Don't show' CTA functionality working as per AC.
- Above issue was due to space copied with timestamp.
- Verified all above issue 1 to 4 are resolved now.

_1) Font family for the banner title is 'Google Sans Display' in Figma where as under actual implementation it is 'Google Sans Text'._ - **_PASS_**

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/3d3a6f9f-dfe1-454c-bc22-36cd40f25a3b)

</details>

2) We don't have figma design for large tablet viewports but design under Viewports having width between 961px to 1132px is looking off with space appearing below the image. - **_PASS_**

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/2c12b52a-23b8-4287-a79d-f62dcafe3d74)

![image](https://github.com/google/site-kit-wp/assets/94359491/9c6306ac-a47b-4fb0-bfb8-28717756fafe)

</details>

_3) In smaller tablet viewports font size is different from figma. On smaller tablets we implemented same font size as mobile viewports. Tested on iPad air and Surface Pro smaller tablet viewports._  **_- PASS_**

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/02fc678e-c161-4d28-a1e4-cebe90413ada)


![image](https://github.com/google/site-kit-wp/assets/94359491/89a6d2d8-70e0-444d-afb3-b2347476fbe6)

</details>

_4) The 'Maybe later' CTA is not functioning as expected according to the acceptance criteria and QA brief. Clicking on the 'Maybe later' CTA does not dismiss the banner but instead renames the CTA to "Don’t show again". Clicking on 'Don’t show again' dismisses the banner. Additionally, upon each refresh when the dashboard loads, the Ads module setup CTA banner briefly displays for a few milliseconds before hiding._ **_PASS_**

<details>

'Maybe later' CTA 

https://github.com/google/site-kit-wp/assets/94359491/f8689ab2-b5aa-4275-8088-83e99e878e1d

'Don't Show'  CTA

https://github.com/google/site-kit-wp/assets/94359491/9b5e1c24-12a0-4561-98b4-8ec41e6be719

</details>
